### PR TITLE
Rewrite to be typescript

### PR DESCRIPTION
**Pull request by supachris28**

### CODE REVIEW

**Review by ukmadlz — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -9,12 +9,11 @@
...
   "scripts": {
-    "lint": "eslint ./src",
-    "lint:fix": "eslint ./src --fix",
-    "unit": "./node_modules/.bin/mocha --require @babel/register --timeout 5s",
-    "coverage": "nyc npm run unit",
-    "test": "npm run lint && npm run unit && npm run coverage",
-    "build": "babel src -d dist",
+    "lint": "tslint --project ./",
+    "lint:fix": "tslint --project ./ --fix",
+    "unit": "nyc --require ts-node/register --require source-map-support/register mocha --recursive 'test/**/*.spec.ts'",
```
Comment (supachris28) [inline package.json]: This can be simplified

File: package.json [hunk 2]
```diff
@@ -32,39 +31,48 @@
...
-    "@packt/error-custom": "0.1.3",
-    "@packt/serverless-responses": "0.3.3",
-    "joi": "14.3.1",
-    "noop2": "2.0.0",
-    "pg": "7.11.0",
-    "pg-hstore": "2.3.3",
-    "pgpass": "1.0.2",
-    "sequelize": "5.9.1"
+    "@hapi/joi": "^15.1.0",
+    "@packt/error-custom": "^0.1.3",
```
Comment (supachris28) [inline package.json]: Bump

File: src/jsonParse.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import ErrorCustom from '@packt/error-custom';
+import errorCustom from '@packt/error-custom';
```
Comment (supachris28) [inline src/jsonParse.ts]: Undo lower

File: src/pagination-options-interface.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export default interface IPaginationOptions {
```
Comment (supachris28) [inline src/pagination-options-interface.ts]: More to interfaces folder

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
+  "compilerOptions": {
+    "lib": [
+      "es2017"
+    ],
+    "module": "commonjs",
+    "target": "es2015",
+    "noImplicitAny": false,
```
Comment (supachris28) [inline tsconfig.json]: Check if this can be true

File: tslint.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (supachris28) [inline tslint.json]: Remove and switch back to selling only

File: src/service-model.ts [hunk 7]
```diff
@@ -0,0 +1,187 @@
+import joi from "@hapi/joi";
+import sequelize from "sequelize";
+import { URL } from "url";
+
+import dbConfigInterface from "./db-config-interface";
```
Comment (supachris28) [inline src/service-model.ts]: IDbConfig

File: src/pagination-options-interface.ts (hunk 4 above)
Comment (supachris28) [reply src/pagination-options-interface.ts]: Move

**Review by ukmadlz — APPROVED** [review]

**Review by mariuslazar93 — APPROVED** [review]

**Pull request merged**